packages/fxa-settings/src/components/ThirdPartyAuth/index.tsx

```diff
@@ -143,15 +143,24 @@ const ThirdPartySignInForm = ({
   const { logViewEventOnce } = useMetrics();
   const stateRef = useRef<HTMLInputElement>(null);
 
-  function onClick() {
+  async function onClick() {
     logViewEventOnce(`flow.${party}`, 'oauth-start');
 
-    if (viewName === 'signin' && party === 'google') {
-      GleanMetrics.thirdPartyAuth.startGoogleAuthFromLogin();
-    } else if (viewName === 'signin' && party === 'apple') {
-      GleanMetrics.thirdPartyAuth.startAppleAuthFromLogin();
+    switch (`${party}-${viewName}`) {
+      case 'google-signin':
+        GleanMetrics.thirdPartyAuth.startGoogleAuthFromLogin();
+        break;
+      case 'apple-signin':
+        GleanMetrics.thirdPartyAuth.startAppleAuthFromLogin();
+        break;
+      case 'google-signup':
+        GleanMetrics.thirdPartyAuth.startGoogleAuthFromReg();
+        break;
     }
 
+    // wait for all the Glean events to be sent before the page unloads
+    await GleanMetrics.isDone();
```

Comment:
@jonalmeida There's an async function to await for all the queued events to be sent.